src/pipelines/rust_app.rs

```diff
@@ -352,15 +352,28 @@ pub struct RustAppOutput {
 
 impl RustAppOutput {
     pub async fn finalize(self, dom: &mut Document) -> Result<()> {
+        let (base, js, wasm, head) = (&self.cfg.public_url, &self.js_output, &self.wasm_output, "html head");
+
+        let with_preload = format!(
+            r#"
+<link rel="preload" href="{base}{wasm}" as="fetch" type="application/wasm" crossorigin>
+<link rel="preload" href="{base}{js}" as="script" type="text/javascript" crossorigin>{head}"#,
+            base = base,
+            js = js,
+            wasm = wasm,
+            head = dom.select(head).html(),
+        );
+        dom.select(head).set_html(with_preload);
```

Comment:
```suggestion
        dom.select(head).append_html(with_preload);
```

The `set_html` method will overwrite the content of the selected element, where as `append_html` will append the given HTML to the end of the selected element.

- https://docs.rs/nipper/0.1.9/nipper/struct.Selection.html#method.set_html
- https://docs.rs/nipper/0.1.9/nipper/struct.Selection.html#method.append_html

Was `append_html` not working as needed?

Reply:
Yeah, the idea is to get the preload at the top, so it doesn't make sense to append it at the bottom.

Reply:
Hmm ... I must not be communicating this very clearly. We can't completely overwrite the contents of the `head` element. What about the title? What about other statically linked resources? Just going to blow them away and call it good? Obvs no `:)`.

However, notwithstanding the issue of blowing away the content, when dealing with the HTML head, having the assets declared at the bottom should make NO difference. The browser will load all of the assets declared therein, and will only execute JS and WASM after said resources have been loaded. So specifically within the HTML head, the ordering should not matter.

In this case, the `preload` directive helps us because of the way our JS WASM loader is declared, so preloading ensures that the assets are in place before the script is executed which would then cause the assets to be loaded. With `preload`, the assets will already be available (as you have pointed out and is the reason for this PR).

Reply:
Oh, we're not overwriting anything. The original head is still there, in the `{head}` block. So we're just prepending.

I wasn't aware the preloading statements could be put further down, so that's an option, but I still think it makes semantically more sense to put them at the top.